// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  // Sends another greeting
  rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}
// クライアントからきたメッセージを保存
  rpc SendRequest (MessageRequest) returns (RequestReply) {}
// メッセージをクライアントに送信
  rpc GetReply (ReplyRequest) returns (MessageReplyList) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

// メッセージを送る
message MessageRequest {
  string name = 2;
  string message = 3;
}
// メッセージを送った結果
message RequestReply {
  string message = 1;
}

// メッセージの受信をリクエストする
message ReplyRequest {
  int32 date = 1;
}

// メッセージを受信する
message MessageReply {
  int32 date = 1;
  string name = 2;
  string message = 3;
}

message MessageReplyList {
  repeated MessageReply messages = 1;
}